src/Controls/src/Core/VisualElement/VisualElement.cs

```diff
@@ -1667,6 +1715,13 @@ void IPropertyPropagationController.PropagatePropertyChanged(string propertyName
 		protected void RefreshIsEnabledProperty() =>
 			this.RefreshPropertyValue(IsEnabledProperty, _isEnabledExplicit);
 
+		/// <summary>
+		/// This method must always be called if some event occurs and the value of
+		/// the <see cref="IsVisibleCore"/> property will change.
+		/// </summary>
+		protected void RefreshIsVisibleProperty() =>
```

Comment:
For a service release, maybe this should be `protected internal` for now. In net9.0 we can make it public.

Reply:
I think it is a good idea